corim/profiles.go

```diff
@@ -0,0 +1,326 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package corim
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/veraison/corim/comid"
+	"github.com/veraison/corim/extensions"
+	"github.com/veraison/eat"
+	"github.com/veraison/go-cose"
+)
+
+// SignedCorimMapExtensionPoints is a list of extension.Point's valid for a
+// SignedCorim.
+var SignedCorimMapExtensionPoints = []extensions.Point{
+	ExtSigner,
+	ExtUnsignedCorim,
+	ExtEntity,
+}
+
+// UnsignedCorimMapExtensionPoints is a list of extension.Point's valid for a
+// UnsignedCorim.
+var UnsignedCorimMapExtensionPoints = []extensions.Point{
+	ExtUnsignedCorim,
+	ExtEntity,
+}
+
+// ComidMapExtensionPoints is a list of extension.Point's valid for a comid.Comid.
+var ComidMapExtensionPoints = []extensions.Point{
+	comid.ExtComid,
+	comid.ExtEntity,
+	comid.ExtTriples,
+	comid.ExtReferenceValue,
+	comid.ExtReferenceValueFlags,
+	comid.ExtEndorsedValue,
+	comid.ExtEndorsedValueFlags,
+}
+
+// AllExtensionPoints is a list of all valid extension.Point's
+var AllExtensionPoints = make(map[extensions.Point]bool) // populated inside init() below
+
+// UnmarshalSignedCorimFromCBOR unmarshals a SignedCorim from provided
+// CBOR data. If there are extensions associated with the profile specified by
+// the data, they will be registered with the UnsignedCorim before it is
+// unmarshaled.
+func UnmarshalSignedCorimFromCBOR(buf []byte) (*SignedCorim, error) {
```

Comment:
The names sounds duplicate to the similar functions given in the unsignedcorim.go `FromCBOR()` Method. Perhaps, we should clarify and also if possible re-name them to remove the confusion.

Reply:
The similarity is the point, as they are parallels of each other -- one is a function, the other is a method. You use one when you don't already have an instance to unmarshal into, and they other when you do.